### DISCUSSION TITLE
iOS15 out_of_memory

### DISCUSSION THREAD

**Discussion by Article19** [pixijs/pixijs#8000] [Q&A]
One of our games suddenly started kicking out a WebGL out_of_memory error in iOS15 (and iOS15.1), in Safari and Chrome (did not test other browsers). This was never an issue in iOS <= 14.x. This happens with PIXI v5 & v6. I was able to track it down to a runtime-generated sprite-sheet with a resolution of 4096x4096 (this is in addition to a few 1024x1024 external sheets and a few bitmap fonts that get loaded). When I reduced our runtime texture dimensions to 3840x3840 the problem went away. I'm quite sure the hardware supports 4096x4096 textures...so not sure if this is a PIXI issue, a browser issue, a device issue, or something else...but thought I would drop a mention here in case anyone else discovers the same thing.

**Comment by eXponenta** [comment]
Related #7928 
This hard ios15 degradation.
You not alone with rhis problem..This is global issues.

Only way - strictly limit textures count and size, limit usage of Text elements, because it create a lot of canvases.
Preload your assets to GPU via prepare plugin. This can help reduce spam to GPU.

There aren't easy in-design solutions.
